### PR TITLE
Replace public config with rpc

### PR DESCRIPTION
**Pull request by andrepimenta**
**Description**

**Should only be merged after https://github.com/MetaMask/metamask-mobile/pull/1970**

This PR replaces public config with rpc: Introduces `unlockStateChanged` and `chainChanged` notifications and `wallet_getProviderState` rpc method and removes public config.

### CODE REVIEW

**Review by ibrahimtaveras00 — APPROVED** [review]
QA Passed 👍

File: app/components/Views/BrowserTab/index.js [hunk 1]
```diff
@@ -688,7 +691,7 @@ export const BrowserTab = props => {
 	};
 
 	/**
-	 * Disabling iframes for now
+	 * Disabling iframes for now´
```
Comment (rekmarks) [inline app/components/Views/BrowserTab/index.js:694]: Stray back-tick:
```suggestion
	 * Disabling iframes for now
```

**Review by rekmarks — CHANGES_REQUESTED** [review]
This implementation LGTM! However, before merging this, https://github.com/MetaMask/mobile-provider/pull/41 must be merged, and `mobile-provider` must be updated with `@metamask/inpage-provider@7.0.0`, once that's published. Then `mobile-provider` needs to be updated here as well, probably in this PR.

**Comment by rekmarks** [comment]
#2114 needs to be merged into this first.

**Comment by rekmarks** [comment]
Since QA passed, we now need to publish `mobile-provider@2.0.0` and add it to this PR, then this can be merged.

File: app/components/Views/BrowserTab/index.js [hunk 2]
```diff
@@ -669,7 +669,24 @@ export const BrowserTab = props => {
 					}, 1500);
 
 					res.result = true;
-				}
+				},
+
+				metamask_getProviderState: async () => {
```
Comment (rekmarks) [inline app/components/Views/BrowserTab/index.js:674]:
```suggestion
				/**
				 * This method is used by the inpage provider to get its state on
				 * initialization.
				 */
				metamask_getProviderState: async () => {
```

**Review by rekmarks — APPROVED** [review]
LGTM!

**Pull request merged**